### DISCUSSION TITLE
WS2812 Drivers

### DISCUSSION THREAD

**Discussion by Witty-Wizard** [stm32duino/Arduino_Core_STM32#2407] [Q&A] [answered]
# Dshot 
I am working on a protocol called Dshot for my project, it is a lot similar to WS2812 LED control signal.
While using the Hal I am able to generate the signal using a DMA and timer. I need help on how to do  the same thing for stm32duino. I look forward to make it into a library for stm boards

**Comment by fpistm** [comment, marked as answer]
Hi @Witty-Wizard 
You can find this section useful: https://github.com/stm32duino/Arduino_Core_STM32/wiki/HAL-configuration#hal-module-only

And this example:
https://www.stm32duino.com/viewtopic.php?t=110

**Reply by Witty-Wizard** [reply]
Thanks!!,
Is there any example with the LL libraries

**Reply by fpistm** [reply]
No. You can use it by simply include the one you need. To be generic we provide a wrapper on header names:
https://github.com/stm32duino/Arduino_Core_STM32/tree/main/libraries/SrcWrapper/inc/LL
